Source/WebCore/rendering/svg/SVGTextLayoutEngine.cpp

```diff
@@ -266,17 +268,17 @@ void SVGTextLayoutEngine::finalizeTransformMatrices(Vector<InlineIterator::SVGTe
     if (textBoxes.isEmpty())
         return;
 
-    for (auto textBox : textBoxes) {
-        auto textBoxTransformation = m_chunkLayoutBuilder.transformationForTextBox(textBox);
-        if (textBoxTransformation.isIdentity())
-            continue;
+    auto boxesToTransform = textBoxes | std::views::filter([&](const auto& textBox) {
+        return !m_chunkLayoutBuilder.transformationForTextBox(textBox).isIdentity();
+    });
 
-        auto it = m_fragmentMap.find(makeKey(*textBox));
-        if (it != m_fragmentMap.end()) {
-            for (auto& fragment : it->value) {
+    for (const auto& textBox : boxesToTransform) {
+        auto textBoxTransformation = m_chunkLayoutBuilder.transformationForTextBox(textBox);
+        if (auto it = m_fragmentMap.find(makeKey(*textBox)); it != m_fragmentMap.end()) {
+            std::ranges::for_each(it->value, [&](auto& fragment) {
```

Comment:
I do not know what is the benefit of using `std::ranges::for_each()` over `for()` but I think
```
for (auto& fragment : *fragment)
```
is simpler and more readable.